internal/app/grpcclient/impl.go

```diff
@@ -104,8 +110,9 @@ func (i *invoker) unary(
 	if err != nil {
 		// If an error was returned, convert it to a gRPC error
 		protoErr = grpcutil.ConvertGrpcToProtoError(err)
-	} else {
-		// If the call was successful, get the returned payloads
+	} else if resp.Payload != nil {
```

Comment:
You can change this back to `} else {` since it is not possible for payload to be nil if err is nil. One or the other will be non-nil since a unary RPC _must_ either return a response or an error.